src/viadot/sources/vid_club.py

```diff
@@ -0,0 +1,400 @@
+
+        Raises:
+            ValidationError: If any source different than the ones in the list are used.
+        """
+        headers = self.headers
+        if source not in ["jobs", "product", "company", "survey"]:
+            raise ValidationError(
+                "The source has to be: jobs, product, company or survey"
+            )
+        if to_date is None:
+            to_date = datetime.today().strftime("%Y-%m-%d")
+
+        response, first_url = self.check_connection(
+            source=source,
+            from_date=from_date,
+            to_date=to_date,
+            items_per_page=items_per_page,
+            region=region,
+        )
+
+        if isinstance(response, dict):
+            keys_list = list(response.keys())
+        elif isinstance(response, list):
+            keys_list = list(response[0].keys())
+        else:
+            keys_list = []
+
+        ind = "next" in keys_list
+
+        if "data" in keys_list:
+            df = pd.json_normalize(response["data"])
+            df = pd.DataFrame(df)
+            length = df.shape[0]
+            page = 1
+
+            while length == items_per_page:
+                if ind is True:
+                    next_page = response["next"]
+                    url = f"{first_url}&next={next_page}"
+                else:
+                    page += 1
+                    url = f"{first_url}&page={page}"
+                response_api = handle_api_response(
+                    url=url, headers=headers, method="GET"
+                )
+                response = response_api.json()
+                df_page = pd.json_normalize(response["data"])
+                df_page = pd.DataFrame(df_page)
+                if source == "product":
+                    df_page = df_page.transpose()
+                length = df_page.shape[0]
+                df = pd.concat((df, df_page), axis=0)
+        else:
+            df = pd.DataFrame(response)
+
+        return df
+
+    def to_df(
+        self,
+        if_empty: str = "warn",
+    ) -> pd.DataFrame:
+        """
+        Looping get_response and iterating by date ranges defined in intervals.
+
+        Stores outputs as DataFrames in a list. At the end, daframes are concatenated
+        in one and dropped duplicates that would appear when quering.
+
+        Args:
+            if_empty (str, optional): What to do if a fetch produce no data.
+                Defaults to "warn
+
+        Returns:
+            pd.DataFrame: Dataframe of the concatanated data carried in the responses.
+        """
+        starts, ends = self.intervals(
+            from_date=self.from_date,
+            to_date=self.to_date,
+            days_interval=self.days_interval
+        )
+
+        dfs_list = []
+        if len(starts) > 0 and len(ends) > 0:
+            for start, end in zip(starts, ends):
+                logger.info(f"ingesting data for dates [{start}]-[{end}]...")
```

Comment:
The `Source` base class already has a logger, so you can just do that

```suggestion
                self.logger.info(f"ingesting data for dates [{start}]-[{end}]...")
```

Reply:
Done